src/main/scala/co/teapot/tempest/graph/MemMappedDynamicUnidirectionalGraph.scala

```diff
@@ -59,27 +59,35 @@ class MemMappedDynamicUnidirectionalGraph(allocator: MemoryMappedAllocator,
   //   3) a (12 * nodeCapacity)-byte array of node data
   //        (4-byte degree and an 8-byte neighborPointer per node)
 
-  def dataPointer = data.getLong(dataPointerPointer)
-  private def setDataPointer(p: Long): Unit = data.putLong(dataPointerPointer, p)
+  def dataPointer: Pointer = data.getPointer(dataPointerPointer)
+  private def setDataPointer(p: Pointer): Unit = data.putPointer(dataPointerPointer, p)
 
   // We store maxNodeId as a long to support ids between 2**31 and 2**32.
-  def maxNodeId: Int = maxNodeIdLong.toInt
+  def maxNodeId: Int = {
+    val result = maxNodeIdLong
+    require(result < Int.MaxValue)
+    result.toInt
+  }
   def maxNodeIdLong: Long =
     data.getLong(dataPointer + MaxNodeIdOffset)
   private def setMaxNodeId(id: Long): Unit = {
     data.putLong(dataPointer + MaxNodeIdOffset, id, allocator.syncAllWrites)
   }
 
-  private def nodeCapacity: Long = data.getLong(dataPointer + NodeCapacityOffset)
+  private def nodeCapacity: Int = {
```

Comment:
Why change`nodeCapacity` to be an `Int`? I think it's possible we'll want between 2^31 and 2^32 nodes again in the future.